marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -45,24 +48,23 @@
     /**
      *
      * @param hubClient
-     * @param taskId
      * @param schema
      * @param params contains all the params provided by Spark, which will include all connector-specific properties
      */
-    public HubDataWriter(HubClient hubClient, long taskId, StructType schema, Map<String, String> params) {
+    public HubDataWriter(HubClient hubClient, StructType schema, Map<String, String> params) {
         this.records = new ArrayList<>();
         this.schema = schema;
         this.batchSize = params.containsKey("batchsize") ? Integer.parseInt(params.get("batchsize")) : 100;
 
-        final String apiModulePath = params.containsKey("apipath") ? params.get("apipath") : "/data-hub/5/data-services/ingestion/bulkIngester.api";
-        logger.info("Will write to endpoint defined by: " + apiModulePath);
+        JsonNode endpointParamsJsonNode = determineIngestionEndpointParams(params);
+
         this.loader = InputEndpoint.on(
             hubClient.getStagingClient(),
-            hubClient.getModulesClient().newTextDocumentManager().read(apiModulePath, new StringHandle())
+            hubClient.getModulesClient().newJSONDocumentManager().read(endpointParamsJsonNode.get("apiPath").asText(), new StringHandle())
         ).bulkCaller();
 
-        loader.setWorkUnit(new ByteArrayInputStream(("{\"taskId\":" + taskId + "}").getBytes()));
-        loader.setEndpointState(new ByteArrayInputStream(("{\"next\":" + 0 + ", \"uriprefix\":\"" + params.get("uriprefix") + "\"}").getBytes()));
+        this.loader.setEndpointState(new ByteArrayInputStream((endpointParamsJsonNode.get("endpointState").asText()).getBytes()));
```

Comment:
I'll test this locally, but I didn't think "asText" does what we want here - it's just getting the text of this node, it's not the same as toString. I think we want to use JacksonHandle here instead, but I'm not sure. 

Which makes me realize - we really need a test that uses a custom API path. That way, we know for sure that our custom workUnit and endpointState are passed to our custom endpoint correctly.

Let me handle that test - I want to think a bit about where the modules for this should go. I think they can go under ./marklogic-data-hub/src/test/ml-modules, I just need to confirm that. I'll try to submit that tonight or first thing tomorrow morning. 

In the meantime though, this can go forward so that @SameeraPriyathamTadikonda  can make use of it.